### PR TITLE
refactor: removes double loop of pdml transaction

### PR DESCRIPTION
**Pull request by thiagotnunes**
Refactors the PDML transaction class to have a single loop for resumes / retries.
I tried to simplify as much as possible on the retry logic we were doing on the transaction. 

`<rant>`
Unfortunately with Java 7 it is hard to improve this logic by not using exceptions for that without having to write lots of code. I could not also come up with a good immutable approach that would have better readability.
`</rant>`

### CODE REVIEW

**Comment by thiagotnunes** [comment]
Since this contains a fix for retrying `InternalException - EOS`, we will wait until the spanner team has reviewed the code before merging.
@skuruppu I could also remove this fix, so that we can merge this right away.

**Comment by skuruppu** [comment]
Do you mean that this also contains the change from #360? If that's the case, yeah it would be ideal if we decouple the two fixes.

**Comment by thiagotnunes** [comment]
@skuruppu will remove the fix from here

**Comment by thiagotnunes** [comment]
@skuruppu the fix is removed from this PR

File: google-cloud-spanner/src/main/java/com/google/cloud/spanner/PartitionedDmlTransaction.java [hunk 1]
```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2019 Google LLC
```
Comment (skuruppu) [inline google-cloud-spanner/src/main/java/com/google/cloud/spanner/PartitionedDmlTransaction.java:2]: nit: 2020

Comment (thiagotnunes) [reply google-cloud-spanner/src/main/java/com/google/cloud/spanner/PartitionedDmlTransaction.java:2]: Will fix on the next PR (#360).

**Pull request merged**